api/environments/dynamodb/migrator.py

```diff
@@ -0,0 +1,64 @@
+from django.db.models import Prefetch
+from flag_engine.django_transform.document_builders import (
+    build_identity_document,
+)
+
+from environments.models import Environment
+from features.models import FeatureState
+from features.multivariate.models import MultivariateFeatureStateValue
+
+from .dynamodb_wrapper import DynamoIdentityWrapper
+from .types import DynamoProjectMetadata, ProjectIdentityMigrationStatus
+
+
+class IdentityMigrator:
+    def __init__(self, project_id):
+        self.project_metadata = DynamoProjectMetadata.get_or_new(project_id)
+
+    @property
+    def migration_status(self) -> ProjectIdentityMigrationStatus:
+        if not self.project_metadata.migration_start_time:
+            return ProjectIdentityMigrationStatus.MIGRATION_NOT_STARTED
+        elif (
+            self.project_metadata.migration_start_time
+            and not self.project_metadata.migration_end_time
+        ):
+            return ProjectIdentityMigrationStatus.MIGRATION_IN_PROGRESS
+
+        return ProjectIdentityMigrationStatus.MIGRATION_COMPLETED
+
+    @property
+    def is_migration_done(self) -> bool:
+        migration_status = self.migration_status
+        return migration_status == ProjectIdentityMigrationStatus.MIGRATION_COMPLETED
+
+    @property
+    def can_migrate(self) -> bool:
+        migration_status = self.migration_status
+        return migration_status == ProjectIdentityMigrationStatus.MIGRATION_NOT_STARTED
+
+    def migrate(self):
+        self.project_metadata.start_identity_migration()
+        project_id = self.project_metadata.id
+        identity_wrapper = DynamoIdentityWrapper()
+        with identity_wrapper.table.batch_writer() as batch:
```

Comment:
Perhaps here, rather than exposing the table attribute, we should do something like: 

```python
# IdentityMigrator

    def migrate(self):
        self.project_metadata.start_identity_migration()
        project_id = self.project_metadata.id
        identity_wrapper = DynamoIdentityWrapper()
        identities = Identity.objects.filter(environment__project__id=project_id).prefetch_related(...)
        identity_wrapper.write_identities(identities)
```